org.eclipse.swtchart.extensions/src/org/eclipse/swtchart/extensions/core/BaseChart.java

```diff
@@ -712,7 +712,7 @@ public void showSeries(String selectedSeriesId) {
 		}
 	}
 
-	private void applySeriesSettings(ISeries dataSeries, ISeriesSettings seriesSettings) {
```

Comment:
I'd like to keep it private. Reason:

One could apply different series settings to the data series than stored in the settings map. On select or reset data series settings, the stored settings will be applied, which could lead to confusion. Moreover, by using settings from

IBarSeriesData
ILineSeriesData
IScatterSeriesData

it is enured that the correct settings class is used. When setting this method to public, it's not assured.

Reply:
But what is the difference in calling one of the apply<xyz>SeriesSettings? in fact these delegate to this method, so I don't see any "confusion" here, it just makes it necessary to put the code of applySettings everywhere where you need to apply settings in a generic way.

Reply:
Imagine the method is public and a caller applies specific series settings* (we assume that they are of correct type - otherwise applying the settings would be skipped anyhow). What happens if

public void selectSeries(...) is called?
public void resetSeries(...) is called?

* In case of public, casting the IBarSeriesSettings needs to be validated via instanceof.

Reply:
It would happen the same as if someone has called before the already public ones
`public void applyLineSeriesSettings(ILineSeries lineSeries, ILineSeriesSettings lineSeriesSettings)`
or
`public void applyScatterSeriesSettings(ILineSeries scatterSeries, IScatterSeriesSettings scatterSeriesSettings)`
or
`public void applyBarSeriesSettings(IBarSeries barSeries, IBarSeriesSettings barSeriesSettings)`
the 
`public void applySeriesSettings(ISeries dataSeries, ISeriesSettings seriesSettings) `
only delagates to these methods and includes proper instanceof checks... The onyl thing one might argue is that it could throw some kind of IllegalArgumentException if the settings does not match anything, but this would be a change in current behaviour.

Reply:
You're right. Change to public is accepted.